From: Vyatta Package Maintainers <DL-vyatta-help@att.com>
Date: Tue, 22 Oct 2019 13:14:19 +0100
Subject: Tweak logging behaviour

Reduce exit_handler log severity level to INFO since this is not an error.
netplugd successfully caught a SIGTERM and is exiting normally (likely as part
of systemd shutdown).

If a device is removed, then the resynchronization code that polls devices
would get lots of "device not found" errors when testing for flags. Therefore
suppress the message unless debug has been enabled.
---
 main.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/main.c b/main.c
index 642b640..b2e2a4e 100644
--- a/main.c
+++ b/main.c
@@ -64,14 +64,14 @@ handle_interface(struct nlmsghdr *hdr, void *arg)
     parse_rtattrs(attrs, IFLA_MAX, IFLA_RTA(info), len);
 
     if (attrs[IFLA_IFNAME] == NULL) {
-        do_log(LOG_DEBUG, "No interface name");
-        return 0;
+        do_log(LOG_ERR, "No interface name");
+        return -1;
     }
 
     char *name = RTA_DATA(attrs[IFLA_IFNAME]);
 
     if (!if_match(name)) {
-        do_log(LOG_DEBUG, "%s: ignoring event", name);
+        do_log(LOG_INFO, "%s: ignoring event", name);
         return 0;
     }
 
@@ -140,8 +140,8 @@ static void
 exit_handler(int sig)
 {
     tidy_pid();
-    do_log(LOG_ERR, "caught signal %d - exiting", sig);
-    exit(1);
+    do_log(LOG_INFO, "caught signal %d - exiting", sig);
+    exit(0);
 }
 
 struct child_exit
@@ -198,9 +198,10 @@ poll_interfaces(void)
             return 0;
 
         memcpy(ifr.ifr_name, info->name, sizeof(ifr.ifr_name));
-        if (ioctl(sockfd, SIOCGIFFLAGS, &ifr) < 0)
-            do_log(LOG_ERR, "%s: can't get flags: %m", info->name);
-        else {
+        if (ioctl(sockfd, SIOCGIFFLAGS, &ifr) < 0) {
+            if (debug)
+                do_log(LOG_ERR, "%s: can't get flags: %m", info->name);
+        } else {
             ifsm_flagchange(info, ifr.ifr_flags);
             ifsm_flagpoll(info);
         }
